src/Microsoft.ML.AutoML/API/AutoMLExperimentExtension.cs

```diff
@@ -0,0 +1,101 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.Extensions.DependencyInjection;
+using static Microsoft.ML.DataOperationsCatalog;
+
+namespace Microsoft.ML.AutoML.API
+{
+    public static class AutoMLExperimentExtension
+    {
+        public static AutoMLExperiment SetDataset(this AutoMLExperiment experiment, IDataView train, IDataView test)
+        {
+            var datasetManager = new TrainTestDatasetManager()
+            {
+                TrainDataset = train,
+                TestDataset = test
+            };
+
+            experiment.ServiceCollection.AddSingleton<IDatasetManager>(datasetManager);
+            experiment.ServiceCollection.AddSingleton(datasetManager);
+
+            return experiment;
+        }
+
+        public static AutoMLExperiment SetDataset(this AutoMLExperiment experiment, TrainTestData trainTestSplit)
+        {
+            return experiment.SetDataset(trainTestSplit.TrainSet, trainTestSplit.TestSet);
+        }
+
+        public static AutoMLExperiment SetDataset(this AutoMLExperiment experiment, IDataView dataset, int fold = 10)
+        {
+            var datasetManager = new CrossValidateDatasetManager()
+            {
+                Dataset = dataset,
+                Fold = fold,
+            };
+
+            experiment.ServiceCollection.AddSingleton<IDatasetManager>(datasetManager);
+            experiment.ServiceCollection.AddSingleton(datasetManager);
+
+            return experiment;
+        }
+
+
+        public static AutoMLExperiment SetBinaryClassificationMetric(this AutoMLExperiment experiment, BinaryClassificationMetric metric, string labelColumn = "label", string predictedColumn = "PredictedLabel")
+        {
+            var metricManager = new BinaryMetricManager(metric, predictedColumn, labelColumn);
```

Comment:
>predictedColumn, labelColumn [](http://example.com/codeflow?start=64&length=28)

should we flip the order of these parameters to be consistent with the rest of APIs?

Reply:
`labelColumn` should come before `predictedColumn` in order to be consistent with context.Binary.Evaluation api, I'll update `BinaryMetricManager` though

Reply:
### Resolved